airbyte-webapp/src/components/ApiErrorBoundary/ApiErrorBoundary.tsx

```diff
@@ -27,7 +27,10 @@ class ApiErrorBoundary extends React.Component<unknown, BoundaryState> {
       return { errorId: ErrorId.VersionMismatch, message: error.message };
     }
 
-    if (error.message === "Failed to fetch") {
+    const isNetworkBoundaryMessage = error.message === "Failed to fetch";
+    const is502 = error.status === 502;
```

Comment:
This ErrorBoundary was assumed to keep level of errors processing as low as possible. I think we should probably add `SeverUnavailableError` class and move processing to AirbyteRequestService